cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
 
-The established way of preventing clickjacking involves the use of the header `X-Frame-Options` (see: [Clickjacking_Defense_Cheat_Sheet](/Clickjacking_Defense_Cheat_Sheet "wikilink")). However, CSP 2.0 has a new directive `frame-ancestors`.
+This policy allows images, scripts, AJAX, and CSS from the same origin, and does not allow any other resources to load (eg. object, frame, media, etc).
 
-To prevent all framing of your content use:
+## Mixed Content Policy
 
-`Content-Security-Policy: frame-ancestors 'none'`
+- In order to prevent mixed content (resources being loaded over http, from a document loaded over https), one can use the following directive to block mixed content.
 
-To allow for your site only, use:
+  - `Content-Security-Policy: block-all-mixed-content;`
 
-`Content-Security-Policy: frame-ancestors 'self'`
+- On the other hand, if the developer is migrating from HTTP to HTTPS, the following directive will ensure that all requests will be sent over HTTPS with no fallback to HTTP:
 
-To allow for trusted domain (my-trusty-site.com), do the following:
+  - `Content-Security-Policy: upgrade-insecure-requests;`
 
-`Content-Security-Policy: frame-ancestors my-trusty-site.com `
+If the `upgrade-insecure-requests` is set, the `block-all-mixed-content` is rendered meaningless and should be removed.
 
-A word about support. Not supported in all browsers yet, Chrome 40+ and FF 35+ support, but will also default to X-Frame-Options if it exists. Spec says, CSP should take precedence. <https://w3c.github.io/webappsec/specs/content-security-policy/#frame-ancestors-and-frame-options>
+## Preventing ClickJacking
 
-Also, keep in mind the following (from the [CSP Spec](https://w3c.github.io/webappsec/specs/content-security-policy/#frame-ancestors-and-frame-options)):
+- To prevent all framing of your content use:
+  - `Content-Security-Policy: frame-ancestors 'none';`
+- To allow for your site only, use:
+  - `Content-Security-Policy: frame-ancestors 'self';`
+- To allow for trusted domain , do the following:
+  - `Content-Security-Policy: frame-ancestors trusted.com;`
 
-`The frame-ancestors directive MUST be ignored when monitoring a policy, and when a contained in a policy defined via a meta element.`
+## Loading Files from CDN
 
-In otherwords, this will not work when CSP is in a <meta> tag, and will not work when using Content-Security-Policy-Report-Only.
+The below CSP allows loading from the same origin and loading images and scripts from the CDN:
 
-When a report is generated, the blocked-uri will only have a value if it is the same origin as the page.
+`Content-Security-Policy: default-src 'self'; image-src cdn.example.com; script-src cdn.example.com;`
 
-Refactoring inline code
-=======================
+# Refactoring inline code
 
 By default CSP disables any unsigned JavaScript code placed inline in the HTML source, such as this:
 
+```js
```

Comment:
Replace `js` by `javascript` because if use this to identify JS code in the script generating the alphabetical index.